packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/filtering.ts

```diff
@@ -0,0 +1,141 @@
+import searchX from "@sit-onyx/icons/search-x.svg?raw";
+import { computed, h, ref, toValue, watchEffect, type Ref } from "vue";
+import { createFeature } from "..";
+import { injectI18n } from "../../../../i18n";
+import { removeDiacritics } from "../../../../utils/strings";
+import OnyxMiniSearch from "../../../OnyxMiniSearch/OnyxMiniSearch.vue";
+import type { OnyxMiniSearchProps } from "../../../OnyxMiniSearch/types";
+import OnyxSystemButton from "../../../OnyxSystemButton/OnyxSystemButton.vue";
+import type { DataGridEntry } from "../../types";
+import "./filtering.scss";
+import type { FilterOptions } from "./types";
+
+export const FILTERING_FEATURE = Symbol("Filtering");
+export type FilterState<TEntry> = Partial<Record<keyof TEntry, string | undefined>>;
+export const useFiltering = createFeature(
+  <TEntry extends DataGridEntry>(options?: FilterOptions<TEntry>) => {
+    const filters = ref({}) as Ref<FilterState<DataGridEntry>>;
+    const { t } = injectI18n();
+    const config = computed(() => toValue(options?.columns));
+
+    const isFilterEnabled = computed(() => (col: keyof TEntry) => {
+      return config.value?.[col]?.enabled !== false;
+    });
+
+    const filterData = (entries: Readonly<TEntry>[]) => {
+      return entries.filter((entry) =>
+        Object.entries(filters.value).every(
+          ([column, value]: [keyof TEntry, string | undefined]) => {
+            const columnOptions = config.value?.[column];
+            const filterOptions = { ...options?.filterConfig, ...columnOptions?.config };
+
+            if (value == null || value === "") return true;
+            let searchTerm = value.toString();
+            let entryValue = entry[column]?.toString() ?? "";
+            if (filterOptions?.filterFunc) {
+              return filterOptions.filterFunc(
+                searchTerm,
+                entryValue as TEntry[keyof TEntry],
+                column,
+                entry,
+              );
+            }
+            entryValue = removeDiacritics(entryValue);
+            searchTerm = removeDiacritics(searchTerm);
+            if (!filterOptions.caseSensitive) {
+              entryValue = entryValue.toLowerCase();
+              searchTerm = searchTerm.toLowerCase();
+            }
+            if (filterOptions.trimWhitespace) {
+              entryValue = entryValue.replace(/\s+/g, "");
```

Comment:
```suggestion
              entryValue = entryValue.trim();
```

Why don't we use the native trim method here?